typings/index.d.ts

```diff
@@ -7,20 +7,26 @@ import type {
     GuildMember,
     Message,
     ActionRowBuilder,
-    MessageActionRowComponentData,
     EmbedBuilder,
     MessageMentionOptions,
     MessageReaction,
     PermissionResolvable,
     Snowflake,
     User,
     Awaitable,
-    EmbedData,
     APIEmbed,
     MessageActionRowComponentBuilder,
     GuildTextBasedChannel
 } from 'discord.js';
 
+import type { JSONEncodable } from '@discordjs/builders';
+
+import type {
+    APIActionRowComponent,
+    APIMessageActionRowComponent,
+    APIModalActionRowComponent
+} from 'discord-api-types/v10';
+
```

Comment:
@Nico105 all these types are reexported from djs. So u can just import it from there.

Reply:
alright, I trust you on that :D